test/unit/models/consent.test.ts

```diff
@@ -0,0 +1,174 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+import Config from '../../../config/knexfile'
+import ConsentModel, { Consent } from '../../../src/model/consent'
+
+/*
+ * Mock Consent Resources
+ */
+const partialConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123'
+}
+
+const completeConsent: Consent = {
+  id: '1234',
+  initiatorId: 'pisp-2342-2233',
+  participantId: 'dfsp-3333-2123',
+  credentialId: 123,
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs',
+  credentialPayload: 'dwuduwd&e2idjoj0w'
+}
+
+/*
+ * Consent Resource Model Unit Tests
+ */
+describe('consent', (): void => {
+  let Db: Knex
+  let consentModel: ConsentModel
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+
+    consentModel = new ConsentModel(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  describe('registerConsent', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
```

Comment:
Maybe instead of deleting the whole table, remove only rows added by tests. Jest tests are run in parallel and I am afraid that there is a risk that deleting the table could break other tests that are using it. Maybe this is a reason why sometimes your tests are failing randomly....

Reply:
For the time being, I am adding and deleting the same row for all the tests. So should I change the test data for each test as well? Otherwise, deleting only the row would mimic the same behaviour as deleting the entire table.

Reply:
keeping independent data between tests in the central database is a good practice and makes our life easy when working with tests.

Reply:
For synchronous tests, how can I remove specific rows anyways? I do not know which test will be executed next. Maybe I could delete the record in each test itself instead of using a beforeEach/AfterEach hook. What do you think?

Reply:
If you will keep separate rows for each test then you can delete the whole table in `afterAll`... 
Maybe this article would give some hints ( or make things more complicated :-P ): 
https://walrus.ai/blog/2020/04/testing-database-interactions-with-jest/
Any ideas @lewisdaly @kleyow @sridharvoruganti ?

Reply:
Fair enough. I'll implement the changes.

Reply:
@eoln the walrus link that you shared talks about deleting all the rows in the DB using `beforeEach` hook for each test. So which approach should I follow? Deleting after every test or using different records for each test and deleting everything at the end of all tests?

Reply:
The easy option is to separate everything into its own `describe` block.

The better approach is to keep track of which rows you have created, maybe make random ids, and delete them at the end of each test.

When you're creating models, add them to some list called `cleanup`

You can have a `beforeEach` that iterates through `cleanup` and deletes them 1 at a time.